[PT-br] Update concat() Array Page (#13083)
mdn/translated-content: files/pt-br/web/javascript/reference/global_objects/array/concat/index.md

{{JSRef("Global_Objects", "Array")}}

O método **`concat()`** é utilizado para mesclar dois ou mais arrays.
Esse método não altera os arrays existentes, mas, em vez disso, retorna um novo array.

{{EmbedInteractiveExample("pages/js/array-concat.html","shorter")}}

## Sintaxe

```js-nolint
concat()
concat(valor0)
concat(valor0, valor1)
concat(valor1, valor2, /*...,*/ valorN)
```

### Parâmetros

- `valorN` {{optional_inline}}
  - : Arrays e/ou valores para concatenar em um novo array. Se todos
    os parâmetros `valorN` forem omitidos, `concat` retorna uma

>     [shallow copy](/en-US/docs/Glossary/Shallow_copy) de todos os arrays existentes do qual é chamado. Veja a descrição abaixo

Reviewer: ```suggestion
    [shallow copy](/pt-BR/docs/Glossary/Shallow_copy) de todos os arrays existentes do qual é chamado. Veja a descrição abaixo
```

    para mais detalhes.

### Valor de retorno

Uma nova instância {{jsxref("Array")}}.

## Descrição

O método `concat` cria um novo array. O array irá primeiro ser preenchido pelos elementos no objeto no qual é chamado. Então, para cada argumento, seu valor será concatenado no array — para objetos normais ou primitivos, o próprio argumento se tornará um elemento do array final; para arrays ou objetos tipo array com a propriedade [`Symbol.isConcatSpreadable`](/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Symbol/isConcatSpreadable) definida como um valor verdadeiro, cada elemento do argumento será adicionado independentemente ao array final. O método `concat` não recursa em argumentos de array aninhados.

> O método `concat()` é um [copying method](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#copying_methods_and_mutating_methods). Ele não altera `this` ou quaquer um dos arrays fornecidos como argumentos, mas, em vez disso retorna uma [shallow copy](/en-US/docs/Glossary/Shallow_copy) que contém os mesmos elementos dos arrays originais.

Reviewer: ```suggestion
O método `concat()` é um [copying method](/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array#copying_methods_and_mutating_methods). Ele não altera `this` ou quaquer um dos arrays fornecidos como argumentos, mas, em vez disso retorna uma [shallow copy](/pt-BR/docs/Glossary/Shallow_copy) que contém os mesmos elementos dos arrays originais.
```

> O método `concat()` preserva slots vazios se qualquer um dos arrays de origem for [sparse](/en-US/docs/Web/JavaScript/Guide/Indexed_collections#sparse_arrays).

Reviewer: ```suggestion
O método `concat()` preserva slots vazios se qualquer um dos arrays de origem for [sparse](/pt-BR/docs/Web/JavaScript/Guide/Indexed_collections#sparse_arrays).
```

// resulta em [ 0, { '0': 1, '1': 2, '2': 3, length: 3 }, 1, 2, 3 ]
```

### Usando concat() em arrays sparse

Se qualquer uma das arrays é sparse, a array resultante será também sparse:

```js
console.log([1, , 3].concat([4, 5])); // [1, empty, 3, 4, 5]
console.log([1, 2].concat([3, , 5])); // [1, 2, 3, empty, 5]
```

### Chamando concat() em objetos não array

Se o valor `this` não for um array, ele será convertido em um objeto e então tratado da mesma forma que os argumentos para `concat()`. Nesse caso o valor de retorno é sempre um novo array simples.

```js
console.log(Array.prototype.concat.call({}, 1, 2, 3)); // [{}, 1, 2, 3]
console.log(Array.prototype.concat.call(1, 2, 3)); // [ [Number: 1], 2, 3 ]
const arrayLike = { [Symbol.isConcatSpreadable]: true, length: 2, 0: 1, 1: 2 };
console.log(Array.prototype.concat.call(arrayLike, 3, 4)); // [1, 2, 3, 4]
```

## Especificação

{{Specifications}}

## Compatibilidade com navegadores

{{Compat}}

## Veja também

- [Polyfill of `Array.prototype.concat` in `core-js` with fixes and implementation of modern behavior like `Symbol.isConcatSpreadable` support](https://github.com/zloirock/core-js#ecmascript-array)

> - [Indexed collections](/en-US/docs/Web/JavaScript/Guide/Indexed_collections)

Reviewer: ```suggestion
- [Indexed collections](/pt-BR/docs/Web/JavaScript/Guide/Indexed_collections)
```
